OntologyMapper/src/README.md

```diff
@@ -0,0 +1,253 @@
+
+``` json
+  "Header": {
+    "InputOntologies": [
+      {
+        "Name": "input-ontology-name",
+        "Version": "1.0",
+        "DtdlVersion": "v2"
+      }
+    ],
+    "OutputOntologies": [
+      {
+        "Name": "output-ontology-name",
+        "Version": "2.0",
+        "DtdlVersion": "v3"
+      }
+    ]
+  },
+  "InterfaceRemaps": [
+    {
+      "InputDtmi": "dtmi:source-namespace:source-interface-name;1",
+      "OutputDtmi": "dtmi:target-namespace:target-interface;1"
+    },
+  ],
+  "RelationshipRemaps": [
+    {
+      "InputRelationship": "source-relationship-name",
+      "OutputRelationship": "target-relationship-name"
+    }
+  ],
+  "PropertyProjections": [
+    {
+      "OutputDtmiFilter": "*",
+      "OutputPropertyName": "output-property-name",
+      "InputPropertyNames": [ "input-property-name1", "input-property-name2" ],
+      "IsOutputPropertyCollection": true
+    }
+  ],
+  "FillProperties": [
+    {
+      "OutputDtmiFilter": "*",
+      "OutputPropertyName": "output-property-name",
+      "InputPropertyNames": [ "input-property-name1, input-property-name2" ]
+    }
+  ]
+}
+```
+
+## Interfaces and Classes
+    
+### Interface: IOntologyMappingManager
+
+#### Method: ValidateTargetOntologyMapping
+
+##### Description
+
+Validates that all Output DTMIs listed in the Interface Remaps exist in the target object model
+
+##### Parameters
+
+| Name | Description |
+| --- | --- |
+| targetObjectModel | A dictionary of DTMI to DTEntityInfo mappings which are valid in the target ontology |
+| invalidTargets | An output listing of invalid output mappings in the InterfaceRemaps |
+
+##### Returns
+
+true if all targets are valid, false otherwise</returns>
+
+#### Method: TryGetInterfaceRemapDtmi
+
+##### Description
+
+For a given DTMI from the source ontology, get the DTMI for the target ontology
+
+##### Parameters
+
+| Name | Description |
+| --- | --- |
+| inputDtmi | The DTMI from the source ontology |
+| dtmiRemap | An InterfaceRemap if it exists |
+
+##### Returns
+
+true if a remap exists, false otherwise</returns>
+
+#### Method: TryGetRelationshipRemap
+
+##### Description
+
+For a given relationship from the source ontology, get the relationship for the target ontology
+
+##### Parameters
+
+| Name | Description |
+| --- | --- |
+| inputRelationship | The relationship from the source ontology |
+| relationshipRemap | A relationshipRemap if it exists |
+
+##### Returns
+
+true if a remap exists, false otherwise</returns>
+
+#### Method: TryGetFillProperty
+
+##### Description
+
+In some cases, the contents of one input property may need to be copied to multiple other fields in the target ontology. For instance, if the target ontology requires that the name field always be populated, but the source name field may be null and the description field be more reliable, a chain of fields can be set here so that there is a priority list of fields that will backfill the name field if the input name field is null.
```

Comment:
I'm getting a little lost here.. First we're talking about `one to many`, then there is an example scenario regarding `null checks` which sounds more like `many to one`.